controller/api/public/stat_summary.go

```diff
@@ -57,6 +57,12 @@ type podCount struct {
 }
 
 func (s *grpcServer) StatSummary(ctx context.Context, req *pb.StatSummaryRequest) (*pb.StatSummaryResponse, error) {
+
+	// check for well-formed request
+	if req.GetSelector().GetResource() == nil {
```

Comment:
Is it ever possible for `GetSelector` to return nil? Just to be extra careful, I'd write this as:

```
	if req.GetSelector() == nil || req.GetSelector().GetResource() == nil {
```

Reply:
I believe this is still safe provided we chain the safe `Get*()` methods together. An example:
https://play.golang.org/p/Ns8oAg-ArmT

More info:
https://stackoverflow.com/questions/42238624/calling-a-method-on-a-nil-struct-pointer-doesnt-panic-why-not

Reply:
Ahhh, yep, you're totally right -- carry on.